docs/guides/securing-apps/token-exchange.adoc

```diff
@@ -6,6 +6,294 @@ title="Using token exchange"
+admin console in the tab `Advanced` of the OIDC client. The other available value of the switch is `Same session`, which means that refresh token is allowed just if refresh token can use the same user
+session like the subject token has. In case that subject token is coming from transient session or from link:{adminguide_link}#_offline-access[Offline session], the requesting refresh token will not
+be allowed. Similarly it will not be allowed to request offline token (using `scope=offline_access`).
+
+* Token exchange never creates new link:{adminguide_link}#managing-user-sessions[user session]. In case that `requested_token_type` is refresh token, it may eventually create new client session in the user session
+for the requester client (if client session was not yet created).
+
+* {project_name} Token exchange does not yet have support for the `resource` parameter.
+
+* Token exchange specification mentions concepts of https://datatracker.ietf.org/doc/html/rfc8693#name-delegation-vs-impersonation[impersonation and delegation]. {project_name} has support for the
+impersonation use case, but not yet for the delegation use case.
+
+==== Revocation
+
+Assuming that there is subject token `access-token1` issued to client `initial-client`. Here are some considerations related to token revocation:
+
+* For the case when `access-token1` was exchanged to `access-token2` of client `requester-client`, the revocation of `access-token1` will not revoke `access-token2`. Supporting of "revocation chain" for access
+tokens would mean quite an overhead. So considering this, it is the responsibility of the administrator that access tokens are short-lived and will be revoked automatically after some time.
+
+* For the case when `access-token1` was exchanged to `refresh-token2` of client `requester-client`, we try to support revocation chain. This means that:
+  ** Revocation of `access-token1` will revoke also `refresh-token2`. Moreover this will remove the client session of the client `requester-client` from the
+     user session and hence all refresh tokens of `requester-client` in this user session will be effectively revoked
+  ** In case that `refresh-token2` and it's related access token was used for the further token exchange to different client, then revocation of `access-token1` will revoke those subsequent token exchanges
+     as well. In other words, whole "chain" of exchanged tokens is going to be revoked.
+  ** Note that access token should be valid when the revocation endpoint is invoked. If you do not have valid access token as the original `access-token1` is already expired, you can perhaps use other
+     access token issued to same client in the same user session. The exchanged tokens like `refresh-token2` and others from the "chain" should be revoked.
+
+[[_legacy-token-exchange]]
+== Legacy token exchange
+
```

Comment:
We can add another note [here](https://github.com/keycloak/keycloak/blob/28dab5ef1046dd046a83a13c5be0c923f7679e63/docs/guides/securing-apps/token-exchange.adoc?plain=1#L300-L318) saying that now to use legacy internal to internal token exchange you need to disable the new standard token exchange: `--features token-exchange,admin-fine-grained-authz --features-disabled=token-exchange-standard`. For the migration period.

Reply:
Thanks for noticing this. It is not needed to disable `token-exchange-standard` as if the client does not have `Standard token exchange` switch enabled, it won't serve requests by `StandardTokenExchangeProvider` and will fallback to V1. So people using legacy V1 token exchange do not need to do anything if they still want to stick to V1. In other words, once we enable standard token exchange in `main`, I hope that `StandardTokenExchangeV1Test` will still work the same as before without any changes needed in that test.

But I've rather added this explicitly to the "migration" section as I did not mentioned it before :-) Also added some note to "internal-internal" of V1 token exchange and people should rather use the standard.